apps/mobile/src/hooks/use-pressed-state.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react';
+import { GestureResponderEvent } from 'react-native';
+
+interface UsePressedStateInputProps {
+  onPressIn?: (event: GestureResponderEvent) => void;
+  onPressOut?: (event: GestureResponderEvent) => void;
+}
+
+/**
+ * Access pressed state of interactive elements.
+ * @example Basic example
+ * const { pressed, onPressIn, onPressOut } = usePressedState()
+ * return <Pressable onPressIn={onPressIn} onPressOut={onPressOut}/>
+
+ * @example Pass outer props to make sure incoming onPressIn and onPressOut are invoked.
+ * function Button(props: ButtonProps) {
+ *   const { pressed, onPressIn, onPressOut } = usePressedState(props)
+ *   return <Pressable onPressIn={onPressIn} onPressOut={onPressOut}/>
+ * }
+ * */
+export function usePressedState(props: UsePressedStateInputProps = {}) {
+  const [pressed, setPressed] = useState(false);
+
+  const onPressIn = (event: GestureResponderEvent) => {
+    setPressed(true);
+    props.onPressIn?.(event);
+  };
```

Comment:
```suggestion
  function onPressIn(event: GestureResponderEvent) {
    setPressed(true);
    props.onPressIn?.(event);
  }
```

Reply:
D'uh, old habits. It's my todos to make this lintable.

Reply:
There are loads of new linting features in eslint 9 we haven't gotten around to using yet.